plat/drivers/rtc/pl031.c

```diff
@@ -208,8 +213,77 @@ static const struct rtc_ops ops_pl031 = {
 	.irq_handler		= pl031_irq_handler,
 };
 
-struct rtc_dev rtc_pl031 = {
+static struct rtc_dev rtc_pl031 = {
 	.name	= "rtc_pl031",
 	.id	= 0,
 	.ops	= &ops_pl031,
 };
+
+void rtc_init_pl031(void *dtb)
+{
+	__u64 size;
+	__u32 irq_type, hwirq, trigger_type;
+	int fdt_rtc, ret, index, irq, rc;
+
+	uk_pr_info("Probing RTC...\n");
+	/*
+	 * We choose the first available rtc device in device list as the
+	 * system rtc.
+	 */
+	fdt_rtc = fdt_node_offset_idx_by_compatible_list(dtb, -1,
+			rtc_device_list, &index);
+	if (fdt_rtc < 0) {
+		uk_pr_err("Could not find rtc device!, fdt_rtc is %d\n",
+			fdt_rtc);
+		return;
+	}
+
+	ret = fdt_get_address(dtb, fdt_rtc, 0, &pl031_base_addr, &size);
+	if (ret < 0) {
+		uk_pr_err("Could not get rtc address\n");
+		return;
+	}
+
+	rc = gic_get_irq_from_dtb(dtb, fdt_rtc, 0, &irq_type, &hwirq,
+			&trigger_type);
+	if (rc < 0) {
+		uk_pr_err("Failed to find IRQ number from DTB\n");
+		return;
+	}
+
+	irq = gic_irq_translate(irq_type, hwirq);
+	if (irq < 0 || irq >= __MAX_IRQ) {
+		uk_pr_err("Failed to translate IRQ number\n");
+		return;
+	}
+
+	switch (index) {
```

Comment:
As the rtc_list is discarded, this register op can be removed.

Reply:
done